store/memory/store.go

```diff
@@ -0,0 +1,33 @@
+// The memory store provides a store that can be used for testing and single-threaded applications.
+// DO NOT USE this in a production application where persistence beyond a single application lifecycle is necessary
+// or when there are multiple consumers.
+package memory
+
+import (
+  "fmt"
+  "sync"
+)
+
+func New() *Checkpoint{
```

Comment:
Can we call this “Store”

Reply:
Sure.  FWIW, mysql and ddb call it Checkpoint, so that's why I went with that name, but I agree Store is a better name.

Reply:
yep fair enough; that's a piece of lingering tech-debt. the name of the package was changed a few months ago and there is still some cleanup needed. 👍